fluent-plugin-events/lib/fluent/plugin/in_events.rb

```diff
@@ -138,33 +140,32 @@ def start_watcher_thread
       end
 
       def create_config_map
-        @resource_version = 0
-        @configmap.data = { "resource-version": "#{@resource_version}" }
+        @configmap.data = { "resource-version-#{resource_name}": "#{@resource_version}" }
         @client.public_send("create_config_map", @configmap).tap do |map|
           log.debug "Created config map: #{map}"
         end
       end
 
-      def update_config_map
+      def patch_config_map
         pull_resource_version
-        @configmap.data = { "resource-version": "#{@resource_version}"}
-        @client.public_send("update_config_map", @configmap).tap do |map|
-          log.debug "Updated config map: #{map}"
+        @client.public_send("patch_config_map", "fluentd-config-resource-version", {data: { "resource-version-#{resource_name}": "#{@resource_version}"}}, @deploy_namespace).tap do |map|
+          log.debug "Patched config map: #{map}"
         end
       end
 
       def initialize_resource_version
+        @resource_version = 0
         @configmap = ::Kubeclient::Resource.new
         @configmap.metadata = {
           name: "fluentd-config-resource-version",
-          namespace: "sumologic"
+          namespace: @deploy_namespace
         }
 
         # get or create the config map
         begin
-          @client.public_send("get_config_map", "fluentd-config-resource-version", "sumologic").tap do |resource|
+          @client.public_send("get_config_map", "fluentd-config-resource-version", @deploy_namespace).tap do |resource|
```

Comment:
the config map is created in the `deploy_namespace` namespace, and because config map can have multiple KVPs, each resource will have a KVP for its own resource version.

```
$ kubectl describe configmap fluentd-config-resource-version
Name:         fluentd-config-resource-version
Namespace:    sumologic
Labels:       <none>
Annotations:  <none>

Data
====
resource-version-config_maps:
----
2024903
resource-version-pods:
----
2025332
resource-version-services:
----
2024977
Events:  <none>
```
You can see here in the `Data` field there are 3 KVPs